python/perf-kernels/streamk/streamk_kernel.py

```diff
@@ -2,121 +2,100 @@
 import triton.language as tl
 
 
-@triton.jit()
-def get_new_pid(current_pid, num_cus):
-    # Number of XCDs
-    num_xcds = 8
-    # Number of pids per XCD in the new arrangement
-    pids_per_xcd = num_cus // num_xcds
-    # Compute current XCD and local pid within the XCD
-    xcd = current_pid % num_xcds
-    local_pid = current_pid // num_xcds
-
-    # Calculate new pid based on the new grouping
-    new_pid = xcd * pids_per_xcd + local_pid
-    return new_pid
-
-
-@triton.jit()
-def get_tiles_config(
-    M,
-    N,
-    K,
-    num_cus,
-    BLOCK_SIZE_M: tl.constexpr,
-    BLOCK_SIZE_N: tl.constexpr,
-    BLOCK_SIZE_K: tl.constexpr,
-):
-    total_blocks_M = tl.cdiv(M, BLOCK_SIZE_M)
-    total_blocks_N = tl.cdiv(N, BLOCK_SIZE_N)
-    iters_per_tile = tl.cdiv(K, BLOCK_SIZE_K)
-
-    total_tiles = total_blocks_M * total_blocks_N
-    if num_cus > 0 and total_tiles > num_cus:  # Stream-K
-        total_streamk_tiles = total_tiles % num_cus
-        total_full_tiles = total_tiles - total_streamk_tiles
-        total_streamk_iters = total_streamk_tiles * iters_per_tile
-        # iterations related to full waves
-        streamk_iters_pcu = total_streamk_iters // num_cus
-        # iterations related to last (partial) wave
-        streamk_remainder_iters = total_streamk_iters % num_cus
-
-    else:  # all tiles are computed using classical blocking
-        total_full_tiles = total_tiles
-        total_streamk_tiles = 0
-        streamk_iters_pcu = 0
-        streamk_remainder_iters = 0
-        total_streamk_iters = 0
-
-    return iters_per_tile, total_full_tiles, total_streamk_tiles, streamk_iters_pcu, streamk_remainder_iters
-
-
 @triton.jit()
 def streamk_gemm(
     A,
     B,
     C,
+    bias_ptr,
     P,
     locks,
     M,
     N,
     K,
-    num_cus,
     stride_am,
     stride_ak,
     stride_bk,
     stride_bn,
     stride_cm,
     stride_cn,
+    stride_bias,
     BLOCK_SIZE_M: tl.constexpr,
     BLOCK_SIZE_N: tl.constexpr,
     BLOCK_SIZE_K: tl.constexpr,
     GROUP_SIZE_M: tl.constexpr,
+    NUM_SMS: tl.constexpr,
+    BIAS: tl.constexpr,
     EVEN_K: tl.constexpr,
 ):
     pid = tl.program_id(0)
-    pid = get_new_pid(pid, num_cus)
+    pid = (pid % 8) * (NUM_SMS // 8) + (pid // 8)
```

Comment:
This is not needed for anything but `gfx942`, so we will actually remove this if the arch was `gfx90a`.